dd-sdk-android/src/main/kotlin/com/datadog/android/telemetry/internal/TelemetryEventHandler.kt

```diff
@@ -45,6 +56,30 @@ internal class TelemetryEventHandler(
         writer.write(telemetryEvent)
     }
 
+    override fun onSessionStarted(sessionId: String, isDiscarded: Boolean) {
+        seenInCurrentSession.clear()
+    }
+
+    // region private
+
+    private fun canWrite(event: RumRawEvent.SendTelemetry): Boolean {
+        if (!eventSampler.sample()) return false
+
+        val eventIdentity = event.identity
+
+        if (seenInCurrentSession.contains(eventIdentity)) {
```

Comment:
We can miss only if message which is in the `throwable` is different each time, because identity only takes into account throwable class, but not sure if it will be a frequent case.

Anyway, telemetry logging will evolve over the time - we will see if we need to add more information somewhere, or the opposite - remove telemetry from certain places, because things are not actionable.